lib/issuesyncgithub/github.go

```diff
@@ -12,39 +13,93 @@ import (
 	"golang.org/x/oauth2"
 )
 
-// GitHubClient is a wrapper around the GitHub API Client library we
+// Client is a wrapper around the GitHub API Client library we
```

Comment:
Client is a pretty generic name. Would something like AbstractGitHubClient be better?

Reply:
I don't think that is a typical naming pattern in Go. The typical pattern is that the interface takes on the name of "the thing in general". And implementations take on the name "the thing for specific case". I actually prefer this naming pattern. Putting programming language key words into names is a code smell imo (i.e. something being abstract or an interface should not matter to users of the class/interface if you are using these constructs correctly). Lastly the choice of Client over GitHubClient was to avoid repetition since the package name includes the word github (`issuesyncgithub`) already and these classes are always prefixed with their package name when used outside of the package.

Reply:
Hmmm. It just seems like "Client" could refer to a github client, a gitlab client, a web UI client, or just about any other kind of client. I don't feel strongly about it, but I still think that having the word GitHub in there makes it more immediately clear what this interface is doing.